src/vmm/src/devices/virtio/net/device.rs

```diff
@@ -562,6 +630,7 @@ impl Net {
     fn handle_deferred_frame(&mut self) -> Result<(), DeviceError> {
         if self.rate_limited_rx_single_frame() {
             self.rx_deferred_frame = false;
+            self.rx_buffer.deferred_descriptor = None;
```

Comment:
I think this is incorrect. If we defer packet because of the rate limiter and then go into this function, we lose deferred descriptor and quest is never notified about it. 
I guess we just need to call `finish_frame` here.

Reply:
`finish_frame` is called inside `rate_limited_rx_single_frame`, no?

Reply:
you are correct. In this case we don't need `self.rx_buffer.deferred_descriptor = None;` at all.

Reply:
you are right. I will drop it.

Reply:
done